support/docker/build.sh

```diff
@@ -29,11 +29,136 @@ stderr.printf() {
 	printf "$@" >&2
 }
 
-_nix_docker() {
+
+_nix_container_manager() {
+
+	# Ask if the user wants to proceed as root
+	if (($(id -u) == 0)); 
+	then
+		stderr.printf "\nYou don't need to run this script as root.\n\nLear more on how run docker/podman in rootless mode:\n Docker: https://docs.docker.com/engine/security/rootless/\n Podman: https://github.com/containers/podman/blob/main/docs/tutorials/rootless_tutorial.md\n\n"
+		PS3="Running as root. Do you want to continue?: "
+
+		select yn in "Yes" "No"; do
+			case $yn in
+				Yes) 
+					stderr.printf "\nProceeding as root user\n"
+					break
+					;;
+				No)
+					stderr.printf "\nQuiting the build script...\n"
+					exit 0
+					;;
+			esac
+		done
+	fi
+
+	# TODO: Add selection menu of available boards located in boards dir
+
+	# Find which container manager is available
+
+	declare -a availableManagers=()
+
+	# Check if docker is available
+	if ! command -v docker &> /dev/null;
+	then
+		stderr.printf "Docker not installed\n"
+	else
+		availableManagers+=('docker')
+	fi
+
+
+	# Check if podman is available
+	if ! command -v podman &> /dev/null;
+	then
+		stderr.printf "Podman not installed\n"
+	else
+		availableManagers+=('podman')
+	fi
+
+	# Exit if neither docker or podman is available
+	if [ ${#availableManagers[@]} -eq 0 ]
+	then
+		stderr.printf "\nCan't find docker or podman on your system.\nInstall one of them to continue...\n"
+		exit 1
+	fi
+
+	# Add exit command
+	availableManagers+=('Quit')
+	
+
+	local selectedManager
+	# Use switch statements to select from the available managers
+	stderr.printf "\nAvailable container managers on your system:\n"
+	PS3="Select one of the options or quit: "
+	select manager in ${availableManagers[*]}; do
+		case $manager in
+			docker|podman)
+				stderr.printf "\nSelected:\n $manager\n"
+				selectedManager=$manager
+				break
+				;;
+			Quit)
+				stderr.printf "\nSee you soon!\n"
+				exit 0	
+				;;
+			*)
+				stderr.printf "\nInvalid option\n"
+				;;
+		esac
+	done
+
```

Comment:
If I understand this right, this will *always* stop and ask, right?

This would break the current semantics of acting like a "thick" wrapper to `nix-build` if I got this right.

Reply:
I have to check how to expand this functionality.

I'm thinking of "interactive" mode where the user can do the selections from above.
But to be executed without any required interaction by default.